substrate/client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -4564,4 +4565,310 @@ mod tests {
+										| PeerState::PendingRequest { .. }
+										| PeerState::Backoff { .. }
+										| PeerState::Poisoned => {
+											unreachable!("no such state is possible")
+										}
+										PeerState::Disabled { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. } | PeerState::Disabled { .. }));
+										}
+										PeerState::DisabledPendingEnable { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. } | PeerState::DisabledPendingEnable { .. }));
+										}
+										PeerState::Enabled { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. } | PeerState::Disabled { .. } | PeerState::Enabled { .. }));
+										}
+										PeerState::Incoming { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. } | PeerState::Disabled { .. } | PeerState::Incoming { .. }));
+										}
+									}
+								} else {
+									// TODO
+								}
+							},
+							FromSwarmE::DialFailure => {
+								if let Some(entry_before) = entry_before {
+									let entry_after = entry_after.unwrap();
+									match entry_before {
+										| PeerState::Requested
+										| PeerState::PendingRequest { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. }));
+										}
+										// otherwise -> same state
+										PeerState::Backoff { .. } => {
+											assert!(matches!(entry_after, PeerState::Backoff { .. }));
+										}
+										PeerState::Poisoned => {
+											assert!(matches!(entry_after, PeerState::Poisoned ));
+										}
+										PeerState::Disabled { .. } => {
+											assert!(matches!(entry_after, PeerState::Disabled { .. }));
+										}
+										PeerState::DisabledPendingEnable { .. } => {
+											assert!(matches!(entry_after, PeerState::DisabledPendingEnable { .. }));
+										}
+										PeerState::Enabled { .. } => {
+											assert!(matches!(entry_after, PeerState::Enabled { .. }));
+										}
+										PeerState::Incoming { .. } => {
+											assert!(matches!(entry_after, PeerState::Incoming { .. }));
+										}
+									}
+								} else {
+									assert!(entry_after.is_none());
+								}
+							},
+						}
+					},
+					FuzzAction::FromHandler(AFromHandler(pid, id, event)) => {
+						let peer_id = pid.0;
+						let conn = ConnectionId::new_unchecked(id);
+						match &event {
+							NotifsHandlerOut::OpenDesiredByRemote { protocol_index } => {
+								let _ = protocol_index;
+							}
+							NotifsHandlerOut::CloseDesired { protocol_index } => {
+								let _ = protocol_index;
+							}
+							NotifsHandlerOut::CloseResult { protocol_index } => {
+								let _ = protocol_index;
+							}
+							// TODO OpenResultOk
+							NotifsHandlerOut::OpenResultErr { protocol_index } => {
+								let _ = protocol_index;
+							}
+							NotifsHandlerOut::Notification { protocol_index, message } => {
+								let _ = protocol_index;
+								let _ = message;
+							}
```

Comment:
Are these still TODO or do they not require logic?

Reply:
Still TODO for on_connection_handler_event checks